connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockActionTranslator.java

```diff
@@ -106,9 +106,25 @@ public void translate(PlayerActionPacket packet, GeyserSession session) {
                 // Handled in BedrockInventoryTransactionTranslator
                 break;
             case START_BREAK:
-                ClientPlayerActionPacket startBreakingPacket = new ClientPlayerActionPacket(PlayerAction.START_DIGGING, new Position(packet.getBlockPosition().getX(),
-                        packet.getBlockPosition().getY(), packet.getBlockPosition().getZ()), BlockFace.values()[packet.getFace()]);
-                session.sendDownstreamPacket(startBreakingPacket);
+                String fireBlock = null;
+                if(session.getConnector().getConfig().isCacheChunks()) {
+                    //check if Chunk Cache is enabled
+                    int fireLocation = session.getConnector().getWorldManager().getBlockAt(session, Vector3i.from(packet.getBlockPosition().getX(),
+                            packet.getBlockPosition().getY() + 1, packet.getBlockPosition().getZ()));
+                    fireBlock = BlockTranslator.getJavaIdBlockMap().inverse().get(fireLocation);
+                }
+                //Check if fire is present to destroy
+                if((fireBlock.contains("minecraft:fire")|| fireBlock.contains("minecraft:soul_fire")) && packet.getFace() == 1) {
```

Comment:
`fireBlock` will be null if chunks are not cached, and hence will throw a NullPointerException here. This should probably be included in the above check since this is only going to work if chunk caching is enabled anyway.